AndrewTehan/hometask-3/cli.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+ENTER_MESSAGE = 'Enter city name: '
+ERROR_MESSAGE = 'City name can not be blank!'
+EXIT_MESSAGE = 'exit!'
+
+# this class contains describes cli for rainfall stats
+class Cli
+  def run_cli
+    loop do
+      input = wait_for_input
+
+      exit if input == EXIT_MESSAGE
+
+      if input.empty?
+        show_error
+      else
+        show_result(input)
+      end
+    end
+  end
+
+  private
+
+  def data
+    @data ||= File.open('data').read
+  end
```

Comment:
It's not critical, but it's better to consider an order of methods' calling and their place in a file

you call `data` only in `show_result`. For some reason, `data` is placed above the method called it in code. So it's highly recommended to take into account such micro details in programming. Jumping through the codebase is distrusting and not so elegant, as Ruby code is supposed to be.